### PR TITLE
Add user affordance for scanner modules that can create a new session

### PR DESCRIPTION
**Pull request by dwelch-r7**
Adding some messaging to the end of scans to tell the user how many credentials and/or sessions were successful
We can potentially flesh this out further in the future by printing out a table of the good creds/created sessions and extending this more broadly to all scanner types but we've limited the scope to just the login modules for the new SMB, and SQL session types for now

# Verification Steps
- [x] Boot up framework
- [x] Use the `smb_login` module
- [x] Perform a single run against a set of hosts with `CreateSession` set to false
- [x] Repeat above but for `CreateSession` set to true
- [x] Verify the output messages are correct (correct number of good creds/sessions etc)
- [x] Repeat for `postgres_login`, `mssql_login` and `mysql_login`

### CODE REVIEW

File: modules/auxiliary/scanner/mssql/mssql_login.rb [hunk 1]
```diff
@@ -56,6 +56,19 @@ def create_session?
...
 
+  def run
+    results = super
+    logins = results.flat_map { |_k, v| v[:successful_logins] }
+    sessions = results.flat_map { |_k, v| v[:successful_sessions] }
+    print_status("Bruteforce completed, #{logins.size} credentials were successful.")
+    if datastore['CreateSession']
+      print_status("#{sessions.size} MSSQL sessions were opened successfully.")
+    else
+      print_status('You can open an MSSQL session with these credentials and CreateSession set to true')
```
Comment (adfoster-r7) [inline modules/auxiliary/scanner/mssql/mssql_login.rb:67]:
```suggestion
      print_status('You can open an MSSQL session with these credentials and %grnCreateSession%clr set to true')
```

(And in the other places)

File: modules/auxiliary/scanner/mssql/mssql_login.rb [hunk 2]
```diff
@@ -56,6 +56,19 @@ def create_session?
     end
   end
 
+  def run
+    results = super
+    logins = results.flat_map { |_k, v| v[:successful_logins] }
+    sessions = results.flat_map { |_k, v| v[:successful_sessions] }
+    print_status("Bruteforce completed, #{logins.size} credentials were successful.")
+    if datastore['CreateSession']
+      print_status("#{sessions.size} MSSQL sessions were opened successfully.")
```
Comment (adfoster-r7) [inline modules/auxiliary/scanner/mssql/mssql_login.rb:65]:
```suggestion
      print_status("#{sessions.size} MSSQL #{sessions.size == 1 'session was' ? 'sessions were'} opened successfully.")
```

(And for the others)

**Comment by cgranleese-r7** [comment]
Looks good to me 👍 

## MySQL
Tested against `mariadb:11.2.2`:
image

## MSSQL
Tested against `mcr.microsoft.com/mssql/server:2022-preview-ubuntu-22.04`:
image

## PostgreSQL
Tested against `postgres:16.1`:
image

## SMB
Tested against Windows 10 with multiple valid logins:
image

**Review by cgranleese-r7 — APPROVED** [review]

**Comment by cgranleese-r7** [comment]
## Release Notes
This PR update several login modules to now display some messaging to the end of scans to tell the user how many credentials and/or sessions were successful.

**Pull request merged**